Cambios en como se leen las keys y agrega TLS

nbarrera opened the pull request:
* mejora la imagen (a lo CSI) de docker usando alpine linux y actualizando paquetes
* se cambia de donde se obtienen los cetiicados y claves, ahora siempre es desde un directorio
* se habilita TLS con un cetificado auto firmado
* se agrega un endpoint para obtener la clave publica de la firma jwt
* ⚠️  se actualiza la collection de postman, uso una variable para la url... sirve para pasar de http a https pero tambien si lo corro local en el 8080 o si lo corro en la docker instance y mapeo  a otro puerto (ej: `-p 8081:8080`)

nbarrera committed:
no git on docker

nbarrera committed:
ignore certs and keys on git

they will be sent by mail

nbarrera committed:
hardening docker instance

nbarrera committed:
cambio de donde se obtienen las claves para el jwt

la clave con la que se firma el jwt antes se ponia como env var

pero ya no

ahora se ponene en los directorios
keys

nbarrera committed:
habilito HTTPS en el puerto 8080

agrego documetnacion

y agrego una api nueva para obtener la clave publica de la firma del JWT

angelraver commented on auth/KeyMgr.js:
```diff
@@ -0,0 +1,46 @@
+      this.certkey = readFileSync(CERT_KEY_PATH, { encoding: 'utf-8' })
+      console.info(`Cert and cert key ok.`)
+    } catch(e) {
+      throw new Error(`ERROR: Reading public and private keys files. Those should be on the ./keys directory with a specific name check README.md and keys/README.md\nerr: ${e.message}\nstack: ${e.stack}`)
```
Este error tan descriptivo no debería existir en un entorno productivo.

nbarrera replied:
hm, si es verdad...
pasa que es medio para que el que esta haciendo "el deploy" entienda porque no arranca

devops ux ponele jeje

lo podemos ofuscar, pero pierde el sentido me parece... la onda es que no tenga problemas para levantarlo y si tiene problemas que los pueda resolver.

la otra opcion podria ser ofuscarlo/quitarlo y documentar mas 👍

nbarrera committed:
Update README.md

nbarrera committed:
traducir readme md

nbarrera committed:
cambiando verbosidad de error por docs

nbarrera committed:
validacion de input hecha en api.yml

validacion de que no se puede cambiar modificar el id de un course

hecha en el updateCourse

se le hace un delete course.id por las dudas

angelraver committed:
Adding course entity (#7)

Co-authored-by: Nicolas Barrera <nbarrera@gmail.com>
